android/app/src/main/java/com/papel/ui/profile/ProfileActivity.java

```diff
@@ -72,8 +90,72 @@ public void onClick(View view) {
                     following = true;
                 }
             }
+        }); */
+
+    }
+
+    private void fetchProfile() {
+        RequestQueue requestQueue = Volley.newRequestQueue(this);
+        String url = Constants.LOCALHOST + Constants.PROFILE + userId;
+
+        StringRequest request = new StringRequest(Request.Method.GET, url,new Response.Listener<String>() {
+            @Override
+            public void onResponse(String response) {
+                try {
+                    JSONObject responseJSON = new JSONObject(response);
+                    String privacy = responseJSON.getString("privacy");
+                    String name = responseJSON.getString("name");
+                    String surname = responseJSON.getString("surname");
+                    boolean isMe = false; // TODO is must be come from the server
+                    JSONArray articleArray = responseJSON.getJSONArray("articles");
+                    for(int i = 0; i<articleArray.length(); i++) {
+                        articles.add(ResponseParser.parseArticle(articleArray.getJSONObject(i)));
+                    }
+                    JSONArray portfolioArray = responseJSON.getJSONArray("portfolios");
+                    for(int i = 0;i<portfolioArray.length(); i++) {
+                        portfolios.add(ResponseParser.parsePortfolio(portfolioArray.getJSONObject(i)));
```

Comment:
Great idea to fetch all profile data via a single function. This adds speed and readability to our code.